classes/tweet.rb

```diff
@@ -1,4 +1,47 @@
 class Tweet
-  attr_accessor :content, :location, :created_at, :zombie, :tweet_id
+  attr_accessor :content, :location, :created_at,
+                        :zombie, :unique_id, :favs, :retweets,
+                        :original_tweet, :retweeted
+
+  # Class level attributes
+  @@tweet_id = 1
+  @@all_tweets = []
+
+  # Run this block on startup
+  def initialize
+    # Assign a unique id to each tweet and increment the counter
+    @unique_id = @@tweet_id
+    @@tweet_id += 1
+    # Create an empty array to track this tweets favs
+    @favs = []
+    # We do track who retweets this tweet, we just count them
+    @retweets = 0
+    # If this tweet is a retweet, the original tweet is stored here.
+    # Otherwise, this is left as false
+    @original_tweet = false
+    # Add this instance to the Class level array of tweets.
+    @@all_tweets << self
+  end
+
+  # Searches for a tweet instance at the Class level based on the Unique ID
+  # Returns either nil or the correct tweet.
+  def self.find_tweet(unique_id)
+    matching_tweet = nil
+    @@all_tweets.each do | tweet |
```

Comment:
Also, why use `.each` when you can just use `.find`? `.each` will just about double or quadruple the amount of code.
```ruby
def self.find_tweet(id)
  @@all_tweets.find { |tweet| tweet.unique_id == id }
end
```